skyreader/plot/plot.py

```diff
@@ -82,101 +79,106 @@ def create_plot(self, result: SkyScanResult, dozoom: bool = False) -> None:
-            # grid_pix = healpy.ang2pix(nside, THETA, PHI)
-            grid_pix = healpy.ang2pix(nside, np.pi/2. - DEC, RA)
-            this_map = np.ones(healpy.nside2npix(nside))*np.inf
-
-            for pixel_data in result.result[f'nside-{nside}']:
-                pixel = pixel_data['index']
-                # show 2*delta_LLH
-                value = 2*pixel_data['llh']
-                if np.isfinite(value):
-                    if np.isnan(min_llh) or value < min_llh:
-                        minCoDec, min_ra = healpy.pix2ang(nside, pixel)
-                        min_dec = np.pi/2 - minCoDec
-                        min_llh = value
-                    if np.isnan(max_llh) or value > max_llh:
-                        max_llh = value
-                this_map[pixel] = value
-
-            if grid_map is None:
-                grid_map = this_map[grid_pix]
-            else:
-                grid_map = np.where( np.isfinite(this_map[grid_pix]), this_map[grid_pix], grid_map)
-
-            del this_map
-
-            LOGGER.info(f"Completed map for nside {nside}.")
+        (
+            grid_value, grid_ra, grid_dec, equatorial_map
+        ) = extract_map(
+            result, llh_map, angular_error_floor, remove_min_val=not llh_map
+        )
 
-        # clean up
-        if grid_pix is not None:
-            del grid_pix
+        grid_pix = healpy.ang2pix(max(nsides), np.pi/2. - DEC, RA)
+        plotting_map = equatorial_map[grid_pix]
 
-        if grid_map is None:
-            # create an "empty" map if there are no pixels at all
-            grid_pix = healpy.ang2pix(8, np.pi/2 - DEC, RA)
-            this_map = np.ones(healpy.nside2npix(8))*np.inf
-            grid_map = this_map[grid_pix]
-            del this_map
-            del grid_pix
+        min_value = grid_value[0]  # for probability map, this is actually
+        # the max_value
+        min_dec = grid_dec[0]
+        min_ra = grid_ra[0]
 
-        LOGGER.info(f"min  RA: {min_ra *180./np.pi} deg, {min_ra*12./np.pi} hours.")
+        LOGGER.info(
+            f"min  RA: {min_ra * 180./np.pi} deg, {min_ra*12./np.pi} hours."
+        )
         LOGGER.info(f"min Dec: {min_dec * 180./np.pi} deg")
 
         # renormalize
         if dozoom:
-            grid_map = grid_map - min_llh
-            # max_value = max_value - min_value
-            min_llh = 0.
-            max_llh = 50
-
-        grid_map = np.ma.masked_invalid(grid_map)
+            plotting_map = plotting_map - min_value
+            equatorial_map = equatorial_map - min_value
+            vmin = 0.
+            vmax = 50
+            map_to_plot = plotting_map
+        if llh_map:
+            cmap = self.PLOT_COLORMAP
+            text_colorbar = r"$-2 \ln(L)$"
+            vmin = np.nanmin(equatorial_map)
+            vmax = np.nanmax(equatorial_map)
+            map_to_plot = plotting_map
+        else:
+            cmap = matplotlib.colormaps[
+                self.PLOT_COLORMAP.name.split('_')[0]
```

Comment:
```suggestion
                self.PLOT_COLORMAP.name.rstrip('_r')
```

Just bc some matplotlib colormap names can have underscores in them. This string strip will prevent an error with those.